pep-0639.rst

```diff
@@ -1,543 +1,2322 @@
+custom identifiers ``LicenseRef-Public-Domain`` and ``LicenseRef-Proprietary``
+to handle the two common cases where projects have a license, but it is not
+one that has a recognized SPDX license identifier.
+
+For maximum flexibility, custom ``LicenseRef-<CUSTOM-TEXT>`` license
+identifiers could be allowed, which could potentially be useful for niche
+cases or corporate environments where ``LicenseRef-Proprietary`` is not
+appropriate or insufficiently specific, but relying on mainstream Python
+build tooling and the ``License-Expression`` metadata field is still
+desirable to use for this purpose.
+
+This has the downsides, however, of not catching mispellings of the
+canonically defined license identifiers and thus producing license metadata
+that is not a valid match for what the author intended, as well as users
+potentially thinking they have to prepend ``LicenseRef`` in front of valid
+license identifiers, as there seems to be some previous confusion about.
+Furthermore, this encourages the proliferation of bespoke license identifiers,
+which obviates the purpose of enabling clear, unambiguous and well
+understood license metadata for which this PEP was created.
+
+Indeed, for niche cases that need specific, proprietary custom licenses,
+they could always simply specify ``LicenseRef-Proprietary``, and then
+include the actual license files needed to unambiguously identify the license
+regardless (if not using SPDX license identifiers) under the ``License-File``
+fields. Requiring standards-conforming tools to allow custom license
+identifiers does not seem very useful, since standard tools will not recognize
+bespoke ones or know how to treat them. By contrast, bespoke tools, which
+would be required in any case to understand and act on custom identifiers,
+are explicitly allowed, with good reason (thus the ``SHOULD`` keyword)
+to not require that license identifiers conform to those listed here.
+Therefore, this specification still allows such use in private corporate
+environments or specific ecosystems, while avoiding the disadvantages of
+imposing them on all mainstream packaging tools.
+
+As an alternative, a literal ``LicenseRef-Custom`` identifier could be
+defined, which would more explicitly indicate that the license cannot be
+expressed with defined identifiers and the license text should be referenced
+for details, without carrying the negative and potentially inappropriate
+implications of ``LicenseRef-Proprietary``. This would avoid the main
+mentioned downsides (misspellings, confusion, license proliferation) of
+the approve approach of allowing an arbitrary ``LicenseRef``, while
+addressing several of the potential theoretical scenarios cited for it.
+
+On the other hand, as SPDX aims to (and generally does) encompass all
+FSF-recognized "Free" and OSI-approved "Open Source" licenses,
+and and those sources are kept closely in sync and are now relatively stable,
```

Comment:
```suggestion
and those sources are kept closely in sync and are now relatively stable,
```

Reply:
Ah, another one of those...I really should have a regex for that. Fixed, thanks.